pulp_container/app/registry_api.py

```diff
@@ -475,6 +479,161 @@ def get_queryset(self, *args, **kwargs):
+            if key == "repository":
+                req_repository = values[0]
+                if any(v != req_repository for v in values[1:]):
+                    unsatisfiable = True
+                    break
+            elif key == "tag":
+                req_tags.update(values)
+            elif key == "os":
+                req_os = values[0]
+                if any(v != req_os for v in values[1:]):
+                    unsatisfiable = True
+                    break
+            elif key == "architecture":
+                req_architecture = values[0]
+                if any(v != req_architecture for v in values[1:]):
+                    unsatisfiable = True
+                    break
+            elif key.startswith("label:"):
+                if key.endswith(":exists"):
+                    if any(v != "1" for v in values):
+                        # Anything other than "1" causes undefined behavior, so treat it as
+                        # unsatisfiable for simplicity:
+                        unsatisfiable = True
+                        break
+                    label = key[len("label:") : len(key) - len(":exists")]
+                    req_label_exists.add(label)
+                else:
+                    if any(v != values[0] for v in values[1:]):
+                        unsatisfiable = True
+                        break
+                    label = key[len("label:") :]
+                    req_label_values[label] = values[0]
+            else:
+                # This covers unknown query parameters (which we consider as unsatisfiable for
+                # simplicity, rather than causing some error response) as well as any annotation:...
+                # query parameters (whose presence necessarily cause the request to be
+                # unsatisfiable, as this implementation never announces any Annotations):
+                unsatisfiable = True
+                break
+
+        results = []
+
+        if not unsatisfiable:
+            # This part copied from CatalogView:
+            queryset = models.ContainerDistribution.objects.all().only("base_path")
+            distribution_permission = "container.pull_containerdistribution"
+            namespace_permission = "container.namespace_pull_containerdistribution"
+            public_repositories = queryset.filter(private=False)
+            repositories_by_distribution = get_objects_for_user(
+                self.request.user, distribution_permission, queryset
+            )
+            namespaces = models.ContainerNamespace.objects.all()
+            repositories_by_namespace = get_objects_for_user(
+                self.request.user, namespace_permission, namespaces
+            )
+            repositories_by_namespace = queryset.filter(namespace__in=repositories_by_namespace)
+            accessible_repositories = repositories_by_distribution & repositories_by_namespace
+            repositories = (public_repositories | accessible_repositories).distinct()
```

Comment:
s/repositories/registry_repositories

Reply:
<https://github.com/pulp/pulp_container/pull/1301/commits/2818e70ae3b782923b2ed77d3e92bdb37f14ea16> "Only index public repositories" removed this part of the code now